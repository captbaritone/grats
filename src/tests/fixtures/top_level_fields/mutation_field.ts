/** @gqlMutationField */
export function goodbye(): string {
  return "Goodbye, World!";
}
