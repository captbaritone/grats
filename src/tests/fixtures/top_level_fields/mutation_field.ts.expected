-----------------
INPUT
----------------- 
/** @gqlMutationField */
export function goodbye(): string {
  return "Goodbye, World!";
}

-----------------
OUTPUT
-----------------
-- SDL --
type Mutation {
  goodbye: String @metadata(exportName: "goodbye", tsModulePath: "grats/src/tests/fixtures/top_level_fields/mutation_field.ts")
}
-- TypeScript --
import { goodbye as mutationGoodbyeResolver } from "./mutation_field";
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const MutationType: GraphQLObjectType = new GraphQLObjectType({
        name: "Mutation",
        fields() {
            return {
                goodbye: {
                    name: "goodbye",
                    type: GraphQLString,
                    resolve() {
                        return mutationGoodbyeResolver();
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        mutation: MutationType,
        types: [MutationType]
    });
}
